enhancements/20240820-mtu-and-network-information-propagation.md

```diff
@@ -0,0 +1,318 @@
+vlan (int, optional):                           assign VLAN tag. Defaults to none.
+preserveDefaultVlan (boolean, optional):        indicates whether the default vlan must be preserved on the veth end connected to the bridge. Defaults to true.
+vlanTrunk (list, optional):                     assign VLAN trunk tag. Defaults to none.
+enabledad (boolean, optional):                  enables duplicate address detection for the container side veth. Defaults to false.
+macspoofchk (boolean, optional):                Enables mac spoof check, limiting the traffic originating from the container to the mac address of the interface. Defaults to false.
+disableContainerInterface (boolean, optional):  Set the container interface (veth peer inside the container netns) state down. When enabled, IPAM cannot be used.
+```
+
+The above fields are converted to a JSON string and set as a string to k8s CRD NAD object.
+
+```
+cniv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+
+type NetworkAttachmentDefinition struct {
+  metav1.TypeMeta   `json:",inline"`
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+
+  Spec NetworkAttachmentDefinitionSpec `json:"spec"`
+}
+
+type NetworkAttachmentDefinitionSpec struct {
+  Config string `json:"config"`
+}
+
+```
+
+Example: The NAD for Harvester storage network.
+
+```
+kind: NetworkAttachmentDefinition
+spec:
+  config: '{"cniVersion":"0.3.1","type":"bridge","dns":{},"bridge":"storage-br","promiscMode":true,"vlan":1,"ipam":{"type":"whereabouts","range":"10.172.0.0/16"}}'
+  
+```
+
+### Harvester UI sets Fields
+
+When configuring [Vlan Network](https://docs.harvesterhci.io/v1.3/networking/harvester-network#vlan-network) from Harvester UI, below fields are touched:
+
+```
+
+Type: `'L2VlanNetwork', 'UntaggedNetwork'`
+Vlan ID:
+ClusterNetwork: User select.  Implicitly converted to `Bridge` name.
+```
+
+:::note
+
+Harvester only exposes very few fields of Bridge CNI to UI.
+
+:::
+
+
+The `Route` related information is converted to an annotation on NAD.
+
+```
+Route:
+  Automatic
+  Manual:
+    CIDR, Gateway
+```
+
+### User Stories
+
+
+#### Story 1: Install a Cluster with the Predefined MTU
+
+The current implementation is OK.
+
+e.g. MTU 9000
+
+After `Clusternetwork` and `Vlanconfig` are created, all following `NADs` will inherit the MTU value 9000. Pod/VM attached to this NAD also has MTU 9000.
+
+#### Story 2: Change the MTU of a Clusternetwork and Vlanconfig on the Fly
+
+The current implementation is not OK.
+
+Step 1: Install the cluster with the default MTU 1500.
+
+After `Clusternetwork` and `Vlanconfig` is created, all following `NAD`s will inherit the MTU value `1500`.
+
+Step 2: Change the MTU to 9000, step by step on each `Vlanconfig`.
+
+Issue/limitation: MTU on the existing `NAD`s are not updated automatically.
+
+#### Story 3: Set the NAD with a different MTU than the Vlanconfig
```

Comment:
what is benefit/usecase for updating the MTU on NAD directly with a different value other than underlying NIC ? Is it to provide more flexibility to user within the cluster with different vlans ? In that case I think the  UI should also restrict user from configuring MTU greater than what is configured on the underlying NIC.

Reply:
A good point. Such limitation is better to have.